An error occurs when the CDC incrementally synchonizes the bytea field.

## Bug Report

**For English only**, other languages will not accept.

Before report a bug, make sure you have:

- Searched open and closed [GitHub issues](https://github.com/apache/shardingsphere/issues).
- Read documentation: [ShardingSphere Doc](https://shardingsphere.apache.org/document/current/en/overview).

Please pay attention on issues you submitted, because we maybe need more details. 
If no response anymore and we cannot reproduce it on current information, we will **close it**.

Please answer these questions before submitting your issue. Thanks!

### Which version of ShardingSphere did you use?
```
/usr/bin/java
we find java version: java8, full_version=1.8.0_242, full_path=/usr/bin/java
ShardingSphere-5.4.2-SNAPSHOT
Commit ID: dirty-152f56c1195f324d310e794c443bec2b3be7190d
Commit Message: Improve CDCImporter finished flag
Branch: fix
```

### Which project did you use? ShardingSphere-JDBC or ShardingSphere-Proxy?
ShardingSphere-Proxy
shardingsphere-data-pipeline-cdc-client-5.4.2-SNAPSHOT.jar

### Expected behavior
The bytea field is synchronized normally during the incremental cdc synchronization status.

### Actual behavior
An error occurs when the CDC incrementally synchonizes the bytea field.

**the error log of cdc client** 
```
[DEBUG] 2023-11-06 18:15:39.499 [nioEventLoopGroup-2-1] RecordsConsumer - records detail: {"before":[{"name":"id","value":{"@type":"type.googleapis.com/google.protobuf.Empty"}},{"name":"c_int","value":{"@type":"type.googleapis.com/google.protobuf.Empty"}},{"name":"c_smallint","value":{"@type":"type.googleapis.com/google.protobuf.Empty"}},{"name":"c_float","value":{"@type":"type.googleapis.com/google.protobuf.Empty"}},{"name":"c_double","value":{"@type":"type.googleapis.com/google.protobuf.Empty"}},{"name":"c_numeric","value":{"@type":"type.googleapis.com/google.protobuf.Empty"}},{"name":"c_boolean","value":{"@type":"type.googleapis.com/google.protobuf.Empty"}},{"name":"c_char","value":{"@type":"type.googleapis.com/google.protobuf.Empty"}},{"name":"c_text","value":{"@type":"type.googleapis.com/google.protobuf.Empty"}},{"name":"c_bytea","value":{"@type":"type.googleapis.com/google.protobuf.Empty"}},{"name":"c_date","value":{"@type":"type.googleapis.com/google.protobuf.Empty"}},{"name":"c_time","value":{"@type":"type.googleapis.com/google.protobuf.Empty"}},{"name":"c_smalldatetime","value":{"@type":"type.googleapis.com/google.protobuf.Empty"}},{"name":"c_timestamp","value":{"@type":"type.googleapis.com/google.protobuf.Empty"}},{"name":"c_timestamptz","value":{"@type":"type.googleapis.com/google.protobuf.Empty"}},{"name":"c_interval","value":{"@type":"type.googleapis.com/google.protobuf.Empty"}},{"name":"c_array","value":{"@type":"type.googleapis.com/google.protobuf.Empty"}},{"name":"c_json","value":{"@type":"type.googleapis.com/google.protobuf.Empty"}},{"name":"c_jsonb","value":{"@type":"type.googleapis.com/google.protobuf.Empty"}},{"name":"c_uuid","value":{"@type":"type.googleapis.com/google.protobuf.Empty"}},{"name":"c_hash32","value":{"@type":"type.googleapis.com/google.protobuf.Empty"}},{"name":"c_tsvector","value":{"@type":"type.googleapis.com/google.protobuf.Empty"}},{"name":"c_bit","value":{"@type":"type.googleapis.com/google.protobuf.Empty"}},{"name":"c_int4range","value":{"@type":"type.googleapis.com/google.protobuf.Empty"}},{"name":"c_reltime","value":{"@type":"type.googleapis.com/google.protobuf.Empty"}},{"name":"c_abstime","value":{"@type":"type.googleapis.com/google.protobuf.Empty"}},{"name":"c_point","value":{"@type":"type.googleapis.com/google.protobuf.Empty"}},{"name":"c_lseg","value":{"@type":"type.googleapis.com/google.protobuf.Empty"}},{"name":"c_box","value":{"@type":"type.googleapis.com/google.protobuf.Empty"}},{"name":"c_circle","value":{"@type":"type.googleapis.com/google.protobuf.Empty"}},{"name":"c_bitvarying","value":{"@type":"type.googleapis.com/google.protobuf.Empty"}},{"name":"c_cidr","value":{"@type":"type.googleapis.com/google.protobuf.Empty"}},{"name":"c_inet","value":{"@type":"type.googleapis.com/google.protobuf.Empty"}},{"name":"c_macaddr","value":{"@type":"type.googleapis.com/google.protobuf.Empty"}},{"name":"c_hll","value":{"@type":"type.googleapis.com/google.protobuf.Empty"}}],"after":[{"name":"id","value":{"@type":"type.googleapis.com/google.protobuf.Int64Value","value":"223"}},{"name":"c_int","value":{"@type":"type.googleapis.com/google.protobuf.Int32Value","value":-74636000}},{"name":"c_smallint","value":{"@type":"type.googleapis.com/google.protobuf.Int32Value","value":-32217}},{"name":"c_float","value":{"@type":"type.googleapis.com/google.protobuf.FloatValue","value":223.1234}},{"name":"c_double","value":{"@type":"type.googleapis.com/google.protobuf.DoubleValue","value":223.123456}},{"name":"c_numeric","value":{"@type":"type.googleapis.com/google.protobuf.StringValue","value":"223.99"}},{"name":"c_boolean","value":{"@type":"type.googleapis.com/google.protobuf.BoolValue","value":false}},{"name":"c_char","value":{"@type":"type.googleapis.com/google.protobuf.StringValue","value":"223char                         "}},{"name":"c_text","value":{"@type":"type.googleapis.com/google.protobuf.StringValue","value":"223text"}},{"name":"c_bytea","value":{"@type":"type.googleapis.com/google.protobuf.StringValue","value":"\u0000\u0001"}},{"name":"c_date","value":{"@type":"type.googleapis.com/google.protobuf.Timestamp","value":"2023-11-05T16:00:00Z"}},{"name":"c_time","value":{"@type":"type.googleapis.com/google.protobuf.Int64Value","value":"65739348000000"}},{"name":"c_smalldatetime","value":{"@type":"type.googleapis.com/google.protobuf.Timestamp","value":"1999-12-31T16:00:00Z"}},{"name":"c_timestamp","value":{"@type":"type.googleapis.com/google.protobuf.Timestamp","value":"2023-11-06T10:15:39.348Z"}},{"name":"c_timestamptz","value":{"@type":"type.googleapis.com/google.protobuf.Timestamp","value":"2023-11-06T10:15:39.348Z"}},{"name":"c_interval","value":{"@type":"type.googleapis.com/google.protobuf.StringValue","value":"0 years 0 mons 1 days 2 hours 3 mins 4.00 secs"}},{"name":"c_array","value":{"@type":"type.googleapis.com/google.protobuf.StringValue","value":"{1,2,3}"}},{"name":"c_json","value":{"@type":"type.googleapis.com/google.protobuf.StringValue","value":"{\"test\":\"223\"}"}},{"name":"c_jsonb","value":{"@type":"type.googleapis.com/google.protobuf.StringValue","value":"{\"test\": \"223\"}"}},{"name":"c_uuid","value":{"@type":"type.googleapis.com/google.protobuf.StringValue","value":"035e5637-57d2-4ec3-8a4a-785a624e7e63"}},{"name":"c_hash32","value":{"@type":"type.googleapis.com/google.protobuf.StringValue","value":"115f89503138416a242f40fb7d7f338e"}},{"name":"c_tsvector","value":{"@type":"type.googleapis.com/google.protobuf.Empty"}},{"name":"c_bit","value":{"@type":"type.googleapis.com/google.protobuf.StringValue","value":"0000"}},{"name":"c_int4range","value":{"@type":"type.googleapis.com/google.protobuf.StringValue","value":"[1,1000)"}},{"name":"c_reltime","value":{"@type":"type.googleapis.com/google.protobuf.StringValue","value":"1 year 1 mon 10 days"}},{"name":"c_abstime","value":{"@type":"type.googleapis.com/google.protobuf.StringValue","value":"2000-01-02 08:00:00+08"}},{"name":"c_point","value":{"@type":"type.googleapis.com/google.protobuf.StringValue","value":"(1,1)"}},{"name":"c_lseg","value":{"@type":"type.googleapis.com/google.protobuf.StringValue","value":"[(0,0),(2,2)]"}},{"name":"c_box","value":{"@type":"type.googleapis.com/google.protobuf.StringValue","value":"(3,3),(1,1)"}},{"name":"c_circle","value":{"@type":"type.googleapis.com/google.protobuf.StringValue","value":"\u003c(5,5),5\u003e"}},{"name":"c_bitvarying","value":{"@type":"type.googleapis.com/google.protobuf.Empty"}},{"name":"c_cidr","value":{"@type":"type.googleapis.com/google.protobuf.StringValue","value":"192.168.0.0/16"}},{"name":"c_inet","value":{"@type":"type.googleapis.com/google.protobuf.StringValue","value":"192.168.1.1"}},{"name":"c_macaddr","value":{"@type":"type.googleapis.com/google.protobuf.StringValue","value":"08:00:2b:01:02:03"}},{"name":"c_hll","value":{"@type":"type.googleapis.com/google.protobuf.StringValue","value":"\\x484c4c00000000002b05000000000000000000000000000000000000"}}],"metaData":{"database":"sharding_db","schema":"","table":"t_order"},"dataChangeType":"INSERT"}
[ERROR] 2023-11-06 18:15:39.547 [nioEventLoopGroup-2-1] o.a.s.d.p.c.c.h.LoggerExceptionHandler - Socket error:
org.opengauss.util.PSQLException: [20.20.20.52:33416/20.20.20.54:17000] ERROR: invalid byte sequence for encoding "UTF8": 0x00
        at org.opengauss.core.v3.QueryExecutorImpl.receiveErrorResponse(QueryExecutorImpl.java:2922)
        at org.opengauss.core.v3.QueryExecutorImpl.processResults(QueryExecutorImpl.java:2643)
        at org.opengauss.core.v3.QueryExecutorImpl.execute(QueryExecutorImpl.java:375)
        at org.opengauss.jdbc.PgStatement.runQueryExecutor(PgStatement.java:561)
        at org.opengauss.jdbc.PgStatement.executeInternal(PgStatement.java:538)
        at org.opengauss.jdbc.PgStatement.execute(PgStatement.java:396)
        at org.opengauss.jdbc.PgPreparedStatement.executeWithFlags(PgPreparedStatement.java:166)
        at org.opengauss.jdbc.PgPreparedStatement.execute(PgPreparedStatement.java:155)
        at com.zaxxer.hikari.pool.ProxyPreparedStatement.execute(ProxyPreparedStatement.java:44)
        at com.zaxxer.hikari.pool.HikariProxyPreparedStatement.execute(HikariProxyPreparedStatement.java)
        at RecordsConsumer.write(RecordsConsumer.java:101)
        at RecordsConsumer.accept(RecordsConsumer.java:61)
        at RecordsConsumer.accept(RecordsConsumer.java:21)
        at org.apache.shardingsphere.data.pipeline.cdc.client.handler.CDCRequestHandler.processDataRecords(CDCRequestHandler.java:110)
        at org.apache.shardingsphere.data.pipeline.cdc.client.handler.CDCRequestHandler.channelRead(CDCRequestHandler.java:104)
        at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:444)
        at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:420)
        at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:412)
        at io.netty.handler.codec.MessageToMessageDecoder.channelRead(MessageToMessageDecoder.java:103)
        at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:444)
        at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:420)
        at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:412)
        at io.netty.handler.codec.ByteToMessageDecoder.fireChannelRead(ByteToMessageDecoder.java:346)
        at io.netty.handler.codec.ByteToMessageDecoder.channelRead(ByteToMessageDecoder.java:318)
        at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:444)
        at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:420)
        at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:412)
        at io.netty.channel.DefaultChannelPipeline$HeadContext.channelRead(DefaultChannelPipeline.java:1410)
        at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:440)
        at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:420)
        at io.netty.channel.DefaultChannelPipeline.fireChannelRead(DefaultChannelPipeline.java:919)
        at io.netty.channel.nio.AbstractNioByteChannel$NioByteUnsafe.read(AbstractNioByteChannel.java:166)
        at io.netty.channel.nio.NioEventLoop.processSelectedKey(NioEventLoop.java:788)
        at io.netty.channel.nio.NioEventLoop.processSelectedKeysOptimized(NioEventLoop.java:724)
        at io.netty.channel.nio.NioEventLoop.processSelectedKeys(NioEventLoop.java:650)
        at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:562)
        at io.netty.util.concurrent.SingleThreadEventExecutor$4.run(SingleThreadEventExecutor.java:997)
        at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74)
        at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)

```
### Reason analyze (If you can)

### Steps to reproduce the behavior, such as: SQL to execute, sharding rule configuration, when exception occur etc.

### Example codes for reproduce this issue (such as a github link).

--- a/kernel/data-pipeline/dialect/opengauss/src/main/java/org/apache/shardingsphere/data/pipeline/opengauss/ingest/wal/decode/MppdbDecodingPlugin.java
+++ b/kernel/data-pipeline/dialect/opengauss/src/main/java/org/apache/shardingsphere/data/pipeline/opengauss/ingest/wal/decode/MppdbDecodingPlugin.java
@@ -193,9 +193,16 @@ public final class MppdbDecodingPlugin implements DecodingPlugin {
                     throw new DecodingException(ex);
                 }
             case "bytea":
+            case "blob":
                 return decodeBytea(data);
             case "raw":
             case "reltime":
+            case "int4range":
+            case "int8range":
+            case "numrange":
+            case "tsrange":
+            case "tstzrange":
+            case "daterange":
                 return decodePgObject(data, columnType);
             case "money":
                 return decodeMoney(data);
@@ -205,6 +212,7 @@ public final class MppdbDecodingPlugin implements DecodingPlugin {
             case "text":
             case "character":
             case "nvarchar2":
+            case "tsquery":
             default:
                 return decodeString(data).replace("''", "'");
         }
@@ -229,16 +237,8 @@ public final class MppdbDecodingPlugin implements DecodingPlugin {
         }
     }
     
-    private PGobject decodeBytea(final String data) {
-        try {
-            PGobject result = new PGobject();
-            result.setType("bytea");
-            byte[] decodeByte = decodeHex(decodeString(data).substring(2));
-            result.setValue(new String(decodeByte));
-            return result;
-        } catch (final SQLException ignored) {
-            return null;
-        }
+    private Object decodeBytea(final String data) {
+        return decodeHex(decodeString(data).substring(2));
     }
     
     private String decodeMoney(final String data) {

--- a/kernel/data-pipeline/dialect/opengauss/src/test/java/org/apache/shardingsphere/data/pipeline/opengauss/ingest/wal/decode/MppdbDecodingPluginTest.java
+++ b/kernel/data-pipeline/dialect/opengauss/src/test/java/org/apache/shardingsphere/data/pipeline/opengauss/ingest/wal/decode/MppdbDecodingPluginTest.java
@@ -125,7 +125,7 @@ class MppdbDecodingPluginTest {
         assertThat(actual.getLogSequenceNumber(), is(logSequenceNumber));
         assertThat(actual.getTableName(), is("test"));
         Object byteaObj = actual.getAfterRow().get(0);
-        assertThat(byteaObj.toString(), is("1.08"));
+        assertThat(byteaObj, is("1.08"));
     }
     
     @Test
@@ -183,8 +183,8 @@ class MppdbDecodingPluginTest {
         assertThat(actual.getLogSequenceNumber(), is(logSequenceNumber));
         assertThat(actual.getTableName(), is("test"));
         Object byteaObj = actual.getAfterRow().get(0);
-        assertThat(byteaObj, instanceOf(PGobject.class));
-        assertThat(byteaObj.toString(), is(new String(new byte[]{(byte) 0xff, (byte) 0, (byte) 0xab})));
+        assertThat(byteaObj, instanceOf(byte[].class));
+        assertThat(byteaObj, is(new byte[]{(byte) 0xff, (byte) 0, (byte) 0xab}));
     }
     
     @Test
@@ -259,5 +259,49 @@ class MppdbDecodingPluginTest {
         assertTrue(actualLastEvent instanceof CommitTXEvent);
         assertThat(((CommitTXEvent) actualLastEvent).getCsn(), is(3468L));
         assertThat(((CommitTXEvent) actualLastEvent).getXid(), is(1L));
+    }
+    
+    @Test
+    void assertDecodeWithTsrange() {
+        MppTableData tableData = new MppTableData();
+        tableData.setTableName("public.test");
+        tableData.setOpType("INSERT");
+        tableData.setColumnsName(new String[]{"data"});
+        tableData.setColumnsType(new String[]{"tsrange"});
+        tableData.setColumnsVal(new String[]{"'[\"2020-01-01 00:00:00\",\"2021-01-01 00:00:00\")'"});
+        ByteBuffer data = ByteBuffer.wrap(toJSON(tableData).getBytes());
+        WriteRowEvent actual = (WriteRowEvent) new MppdbDecodingPlugin(null).decode(data, logSequenceNumber);
+        Object byteaObj = actual.getAfterRow().get(0);
+        assertThat(byteaObj, instanceOf(PGobject.class));
+        assertThat(byteaObj.toString(), is("[\"2020-01-01 00:00:00\",\"2021-01-01 00:00:00\")"));
+    }
+    
+    @Test
+    void assertDecodeWithDaterange() {
+        MppTableData tableData = new MppTableData();
+        tableData.setTableName("public.test");
+        tableData.setOpType("INSERT");
+        tableData.setColumnsName(new String[]{"data"});
+        tableData.setColumnsType(new String[]{"daterange"});
+        tableData.setColumnsVal(new String[]{"'[2020-01-02,2021-01-02)'"});
+        ByteBuffer data = ByteBuffer.wrap(toJSON(tableData).getBytes());
+        WriteRowEvent actual = (WriteRowEvent) new MppdbDecodingPlugin(null).decode(data, logSequenceNumber);
+        Object byteaObj = actual.getAfterRow().get(0);
+        assertThat(byteaObj, instanceOf(PGobject.class));
+        assertThat(byteaObj.toString(), is("[2020-01-02,2021-01-02)"));
+    }
+    
+    @Test
+    void assertDecodeWithTsquery() {
+        MppTableData tableData = new MppTableData();
+        tableData.setTableName("public.test");
+        tableData.setOpType("INSERT");
+        tableData.setColumnsName(new String[]{"data"});
+        tableData.setColumnsType(new String[]{"tsquery"});
+        tableData.setColumnsVal(new String[]{"'''fff'' | ''faa'''"});
+        ByteBuffer data = ByteBuffer.wrap(toJSON(tableData).getBytes());
+        WriteRowEvent actual = (WriteRowEvent) new MppdbDecodingPlugin(null).decode(data, logSequenceNumber);
+        Object byteaObj = actual.getAfterRow().get(0);
+        assertThat(byteaObj.toString(), is("'fff' | 'faa'"));
     }
 }

--- a/kernel/data-pipeline/scenario/cdc/client/src/main/java/org/apache/shardingsphere/data/pipeline/cdc/client/handler/RetryStreamingExceptionHandler.java
+++ b/kernel/data-pipeline/scenario/cdc/client/src/main/java/org/apache/shardingsphere/data/pipeline/cdc/client/handler/RetryStreamingExceptionHandler.java
@@ -35,7 +35,7 @@ public final class RetryStreamingExceptionHandler implements ExceptionHandler {
     
     private final CDCClient cdcClient;
     
-    private final AtomicInteger maxRetryTimes;
+    private final int maxRetryTimes;
     
     private final int retryIntervalMills;
     
@@ -43,7 +43,7 @@ public final class RetryStreamingExceptionHandler implements ExceptionHandler {
     
     public RetryStreamingExceptionHandler(final CDCClient cdcClient, final int maxRetryTimes, final int retryIntervalMills) {
         this.cdcClient = cdcClient;
-        this.maxRetryTimes = new AtomicInteger(maxRetryTimes);
+        this.maxRetryTimes = maxRetryTimes;
         this.retryIntervalMills = retryIntervalMills;
     }
     
@@ -57,7 +57,7 @@ public final class RetryStreamingExceptionHandler implements ExceptionHandler {
     private void reconnect(final ChannelHandlerContext ctx) {
         retryTimes.incrementAndGet();
         ClientConnectionContext connectionContext = ctx.channel().attr(ClientConnectionContext.CONTEXT_KEY).get();
-        if (retryTimes.get() > maxRetryTimes.get()) {
+        if (retryTimes.get() > maxRetryTimes) {
             log.warn("Stop try to reconnect, stop streaming ids: {}", connectionContext.getStreamingIds());
             connectionContext.getStreamingIds().forEach(each -> CompletableFuture.runAsync(() -> cdcClient.stopStreaming(each)));
             return;

--- a/test/e2e/operation/pipeline/src/test/java/org/apache/shardingsphere/test/e2e/data/pipeline/cases/PipelineContainerComposer.java
+++ b/test/e2e/operation/pipeline/src/test/java/org/apache/shardingsphere/test/e2e/data/pipeline/cases/PipelineContainerComposer.java
@@ -235,7 +235,7 @@ public final class PipelineContainerComposer implements AutoCloseable {
                 .replace("${password}", getPassword())
                 .replace("${url}", getActualJdbcUrlTemplate(storageUnitName, true));
         proxyExecuteWithLog(registerStorageUnitTemplate, 0);
-        long timeout = databaseType instanceof OpenGaussDatabaseType ? 40L : 10L;
+        int timeout = databaseType instanceof OpenGaussDatabaseType ? 60 : 10;
         Awaitility.await().ignoreExceptions().atMost(timeout, TimeUnit.SECONDS).pollInterval(1, TimeUnit.SECONDS).until(() -> showStorageUnitsName().contains(storageUnitName));
     }
     

--- a/test/e2e/operation/pipeline/src/test/java/org/apache/shardingsphere/test/e2e/data/pipeline/cases/PipelineContainerComposer.java
+++ b/test/e2e/operation/pipeline/src/test/java/org/apache/shardingsphere/test/e2e/data/pipeline/cases/PipelineContainerComposer.java
@@ -229,13 +229,14 @@ public final class PipelineContainerComposer implements AutoCloseable {
      * @throws SQLException SQL exception
      */
     public void registerStorageUnit(final String storageUnitName) throws SQLException {
-        String username = getDatabaseType() instanceof OracleDatabaseType ? storageUnitName : getUsername();
+        String username = databaseType instanceof OracleDatabaseType ? storageUnitName : getUsername();
         String registerStorageUnitTemplate = "REGISTER STORAGE UNIT ${ds} ( URL='${url}', USER='${user}', PASSWORD='${password}')".replace("${ds}", storageUnitName)
                 .replace("${user}", username)
                 .replace("${password}", getPassword())
                 .replace("${url}", getActualJdbcUrlTemplate(storageUnitName, true));
         proxyExecuteWithLog(registerStorageUnitTemplate, 0);
-        Awaitility.await().ignoreExceptions().atMost(10, TimeUnit.SECONDS).pollInterval(1, TimeUnit.SECONDS).until(() -> showStorageUnitsName().contains(storageUnitName));
+        int timeout = databaseType instanceof OpenGaussDatabaseType ? 60 : 10;
+        Awaitility.await().ignoreExceptions().atMost(timeout, TimeUnit.SECONDS).pollInterval(1, TimeUnit.SECONDS).until(() -> showStorageUnitsName().contains(storageUnitName));
     }
     
     /**

--- a/test/e2e/operation/pipeline/src/test/java/org/apache/shardingsphere/test/e2e/data/pipeline/cases/PipelineContainerComposer.java
+++ b/test/e2e/operation/pipeline/src/test/java/org/apache/shardingsphere/test/e2e/data/pipeline/cases/PipelineContainerComposer.java
@@ -235,7 +235,7 @@ public final class PipelineContainerComposer implements AutoCloseable {
                 .replace("${password}", getPassword())
                 .replace("${url}", getActualJdbcUrlTemplate(storageUnitName, true));
         proxyExecuteWithLog(registerStorageUnitTemplate, 0);
-        int timeout = databaseType instanceof OpenGaussDatabaseType ? 60 : 10;
+        long timeout = databaseType instanceof OpenGaussDatabaseType ? 40L : 10L;
         Awaitility.await().ignoreExceptions().atMost(timeout, TimeUnit.SECONDS).pollInterval(1, TimeUnit.SECONDS).until(() -> showStorageUnitsName().contains(storageUnitName));
     }
     

--- a/test/e2e/operation/pipeline/src/test/java/org/apache/shardingsphere/test/e2e/data/pipeline/cases/migration/AbstractMigrationE2EIT.java
+++ b/test/e2e/operation/pipeline/src/test/java/org/apache/shardingsphere/test/e2e/data/pipeline/cases/migration/AbstractMigrationE2EIT.java
@@ -76,8 +76,7 @@ public abstract class AbstractMigrationE2EIT {
                 .replace("${ds3}", containerComposer.getActualJdbcUrlTemplate(PipelineContainerComposer.DS_3, true))
                 .replace("${ds4}", containerComposer.getActualJdbcUrlTemplate(PipelineContainerComposer.DS_4, true));
         containerComposer.proxyExecuteWithLog(addTargetResource, 0);
-        long timeout = containerComposer.getDatabaseType() instanceof OpenGaussDatabaseType ? 60L : 10L;
-        Awaitility.await().ignoreExceptions().atMost(timeout, TimeUnit.SECONDS).pollInterval(1L, TimeUnit.SECONDS).until(() -> 3 == containerComposer.showStorageUnitsName().size());
+        Awaitility.await().ignoreExceptions().atMost(10L, TimeUnit.SECONDS).pollInterval(1L, TimeUnit.SECONDS).until(() -> 3 == containerComposer.showStorageUnitsName().size());
     }
     
     protected void createSourceSchema(final PipelineContainerComposer containerComposer, final String schemaName) throws SQLException {

--- a/test/e2e/operation/pipeline/src/test/java/org/apache/shardingsphere/test/e2e/data/pipeline/cases/migration/AbstractMigrationE2EIT.java
+++ b/test/e2e/operation/pipeline/src/test/java/org/apache/shardingsphere/test/e2e/data/pipeline/cases/migration/AbstractMigrationE2EIT.java
@@ -76,7 +76,8 @@ public abstract class AbstractMigrationE2EIT {
                 .replace("${ds3}", containerComposer.getActualJdbcUrlTemplate(PipelineContainerComposer.DS_3, true))
                 .replace("${ds4}", containerComposer.getActualJdbcUrlTemplate(PipelineContainerComposer.DS_4, true));
         containerComposer.proxyExecuteWithLog(addTargetResource, 0);
-        Awaitility.await().ignoreExceptions().atMost(5L, TimeUnit.SECONDS).pollInterval(1L, TimeUnit.SECONDS).until(() -> 3 == containerComposer.showStorageUnitsName().size());
+        long timeout = containerComposer.getDatabaseType() instanceof OpenGaussDatabaseType ? 60L : 10L;
+        Awaitility.await().ignoreExceptions().atMost(timeout, TimeUnit.SECONDS).pollInterval(1L, TimeUnit.SECONDS).until(() -> 3 == containerComposer.showStorageUnitsName().size());
     }
     
     protected void createSourceSchema(final PipelineContainerComposer containerComposer, final String schemaName) throws SQLException {
@@ -104,7 +105,7 @@ public abstract class AbstractMigrationE2EIT {
     
     protected void createTargetOrderTableRule(final PipelineContainerComposer containerComposer) throws SQLException {
         containerComposer.proxyExecuteWithLog(migrationDistSQL.getCreateTargetOrderTableRule(), 0);
-        Awaitility.await().atMost(4L, TimeUnit.SECONDS).pollInterval(1L, TimeUnit.SECONDS).until(() -> !containerComposer.queryForListWithLog("SHOW SHARDING TABLE RULE t_order").isEmpty());
+        Awaitility.await().atMost(10L, TimeUnit.SECONDS).pollInterval(1L, TimeUnit.SECONDS).until(() -> !containerComposer.queryForListWithLog("SHOW SHARDING TABLE RULE t_order").isEmpty());
     }
     
     protected void createTargetOrderTableEncryptRule(final PipelineContainerComposer containerComposer) throws SQLException {

--- a/test/e2e/operation/pipeline/src/test/java/org/apache/shardingsphere/test/e2e/data/pipeline/cases/migration/general/PostgreSQLMigrationGeneralE2EIT.java
+++ b/test/e2e/operation/pipeline/src/test/java/org/apache/shardingsphere/test/e2e/data/pipeline/cases/migration/general/PostgreSQLMigrationGeneralE2EIT.java
@@ -77,7 +77,7 @@ class PostgreSQLMigrationGeneralE2EIT extends AbstractMigrationE2EIT {
             DataSourceExecuteUtils.execute(containerComposer.getSourceDataSource(), containerComposer.getExtraSQLCommand().getFullInsertOrderItem(), dataPair.getRight());
             log.info("init data end: {}", LocalDateTime.now());
             startMigrationWithSchema(containerComposer, SOURCE_TABLE_NAME, "t_order");
-            Awaitility.await().atMost(10L, TimeUnit.SECONDS).pollInterval(1L, TimeUnit.SECONDS).until(() -> listJobId(containerComposer).size() > 0);
+            Awaitility.await().atMost(10L, TimeUnit.SECONDS).pollInterval(1L, TimeUnit.SECONDS).until(() -> !listJobId(containerComposer).isEmpty());
             String jobId = getJobIdByTableName(containerComposer, "ds_0.test." + SOURCE_TABLE_NAME);
             containerComposer.waitIncrementTaskFinished(String.format("SHOW MIGRATION STATUS '%s'", jobId));
             String schemaTableName = String.join(".", PipelineContainerComposer.SCHEMA_NAME, SOURCE_TABLE_NAME);

--- a/test/e2e/operation/pipeline/src/test/java/org/apache/shardingsphere/test/e2e/data/pipeline/cases/task/E2EIncrementalTask.java
+++ b/test/e2e/operation/pipeline/src/test/java/org/apache/shardingsphere/test/e2e/data/pipeline/cases/task/E2EIncrementalTask.java
@@ -56,9 +56,10 @@ public final class E2EIncrementalTask extends BaseIncrementTask {
     private static final List<String> POSTGRESQL_COLUMN_NAMES = Arrays.asList("order_id", "user_id", "status", "t_int2", "t_numeric", "t_bool", "t_bytea", "t_char", "t_varchar", "t_float",
             "t_double", "t_json", "t_jsonb", "t_text", "t_date", "t_time", "t_timestamp", "t_timestamptz");
     
-    private static final List<String> OPENGAUSS_COLUMN_NAMES = Arrays.asList("order_id", "user_id", "status", "c_int", "c_smallint", "c_float", "c_double", "c_numeric", "c_boolean", "c_char",
-            "c_text", "c_bytea", "c_date", "c_time", "c_smalldatetime", "c_timestamp", "c_timestamptz", "c_interval", "c_array", "c_json", "c_jsonb", "c_uuid", "c_hash32", "c_tsvector", "c_bit",
-            "c_int4range", "c_reltime", "c_abstime", "c_point", "c_lseg", "c_box", "c_circle", "c_bitvarying", "c_cidr", "c_inet", "c_macaddr", "c_hll");
+    private static final List<String> OPENGAUSS_COLUMN_NAMES = Arrays.asList("order_id", "user_id", "status", "c_int", "c_smallint", "c_float", "c_double", "c_numeric", "c_boolean",
+            "c_char", "c_text", "c_bytea", "c_raw", "c_date", "c_time", "c_smalldatetime", "c_timestamp", "c_timestamptz", "c_interval", "c_array", "c_json", "c_jsonb", "c_uuid", "c_hash32",
+            "c_tsvector", "c_tsquery", "c_bit", "c_int4range", "c_daterange", "c_tsrange", "c_reltime", "c_abstime", "c_point", "c_lseg", "c_box", "c_circle", "c_bitvarying", "c_cidr", "c_inet",
+            "c_macaddr", "c_hll", "c_money");
     
     private final DataSource dataSource;
     
@@ -132,11 +133,11 @@ public final class E2EIncrementalTask extends BaseIncrementTask {
             LocalDateTime now = LocalDateTime.now();
             String sql = SQLBuilderUtils.buildUpdateSQL(ignoreShardingColumns(OPENGAUSS_COLUMN_NAMES), orderTableName, "?");
             Object[] parameters = {"中文测试", randomInt, random.nextInt(-999, 999), PipelineCaseHelper.generateFloat(), PipelineCaseHelper.generateDouble(), BigDecimal.valueOf(10000),
-                    random.nextBoolean(), "update-char", "update-text", "update-bytea".getBytes(), now.toLocalDate().plusDays(1), now.toLocalTime().plusHours(6), "2023-03-01", now,
+                    random.nextBoolean(), "update-char", "update-text", new byte[]{0, 0, 1}, new byte[]{1, 0}, now.toLocalDate().plusDays(1), now.toLocalTime().plusHours(6), "2023-03-01", now,
                     OffsetDateTime.now(), "1 years 1 mons 1 days 1 hours 1 mins 1 secs", "{4, 5, 6}", PipelineCaseHelper.generateJsonString(1, true), PipelineCaseHelper.generateJsonString(1, false),
-                    UUID.fromString("00000000-000-0000-0000-000000000001").toString(),
-                    DigestUtils.md5Hex(now.toString()), null, "1111", "[1,10000)", "2 years 2 mons 2 days 06:00:00", "2023-01-01 00:00:00+00", "(2.0,2.0)",
-                    "[(0.0,0.0),(3.0,3.0)]", "(1.0,1.0),(3.0,3.0)", "<(5.0,5.0),1.0>", "1010", "192.168.0.0/24", "192.168.1.1", "08:00:3b:01:02:03", null, orderId};
+                    UUID.fromString("00000000-000-0000-0000-000000000001").toString(), DigestUtils.md5Hex(now.toString()), "aaa", "bbb", "1111", "[1,10000)", "[2020-01-02,2021-01-01)",
+                    "[2020-01-01 00:00:00,2021-01-01 00:00:00)", "2 years 2 mons 2 days 06:00:00", "2023-01-01 00:00:00+00", "(2.0,2.0)", "[(0.0,0.0),(3.0,3.0)]", "(1.0,1.0),(3.0,3.0)",
+                    "<(5.0,5.0),1.0>", "1010", "192.168.0.0/24", "192.168.1.1", "08:00:3b:01:02:03", null, 1000, orderId};
             log.info("update sql: {}, params: {}", sql, parameters);
             DataSourceExecuteUtils.execute(dataSource, sql, parameters);
         }

--- a/test/e2e/operation/pipeline/src/test/java/org/apache/shardingsphere/test/e2e/data/pipeline/framework/helper/PipelineCaseHelper.java
+++ b/test/e2e/operation/pipeline/src/test/java/org/apache/shardingsphere/test/e2e/data/pipeline/framework/helper/PipelineCaseHelper.java
@@ -107,13 +107,12 @@ public final class PipelineCaseHelper {
         if (databaseType instanceof OpenGaussDatabaseType) {
             for (int i = 0; i < insertRows; i++) {
                 Object orderId = keyGenerateAlgorithm.generateKey();
-                // TODO openGauss mpp plugin parses single quotes incorrectly
-                result.add(new Object[]{orderId, generateInt(0, 1000), "status" + i, generateInt(-1000, 9999), generateInt(0, 100), generateFloat(), generateDouble(),
+                result.add(new Object[]{orderId, generateInt(0, 1000), "'status'" + i, generateInt(-1000, 9999), generateInt(0, 100), generateInt(-64, 64), generateFloat(), generateDouble(),
                         BigDecimal.valueOf(generateDouble()), false, generateString(6), "texts", "bytea".getBytes(), LocalDate.now(), LocalTime.now(), "2001-10-01",
                         Timestamp.valueOf(LocalDateTime.now()), OffsetDateTime.now(), "0 years 0 mons 1 days 2 hours 3 mins 4 secs", "{1, 2, 3}", generateJsonString(8, false),
-                        generateJsonString(8, true), UUID.randomUUID().toString(), DigestUtils.md5Hex(orderId.toString()), null, "0000", "[1,1000)",
-                        "1 years 1 mons 10 days -06:00:00", "2000-01-02 00:00:00+00", "(1.0,1.0)", "[(0.0,0.0),(2.0,2.0)]", "(3.0,3.0),(1.0,1.0)", "<(5.0,5.0),5.0>", "1111",
-                        "192.168.0.0/16", "192.168.1.1", "08:00:2b:01:02:03", "\\x484c4c00000000002b05000000000000000000000000000000000000"});
+                        generateJsonString(8, true), UUID.randomUUID().toString(), DigestUtils.md5Hex(orderId.toString()), "'rat' 'sat'", "tsquery", "0000", "[1,1000)",
+                        "[2020-01-02,2021-01-01)", "[2020-01-01 00:00:00,2021-01-01 00:00:00)", "1 years 1 mons 10 days -06:00:00", "2000-01-02 00:00:00+00", "(1.0,1.0)", "[(0.0,0.0),(2.0,2.0)]",
+                        "(3.0,3.0),(1.0,1.0)", "<(5.0,5.0),5.0>", "1111", "192.168.0.0/16", "192.168.1.1", "08:00:2b:01:02:03", "\\x484c4c00000000002b05000000000000000000000000000000000000", 999});
             }
             return result;
         }

--- a/test/e2e/operation/pipeline/src/test/java/org/apache/shardingsphere/test/e2e/data/pipeline/framework/helper/PipelineCaseHelper.java
+++ b/test/e2e/operation/pipeline/src/test/java/org/apache/shardingsphere/test/e2e/data/pipeline/framework/helper/PipelineCaseHelper.java
@@ -107,8 +107,8 @@ public final class PipelineCaseHelper {
         if (databaseType instanceof OpenGaussDatabaseType) {
             for (int i = 0; i < insertRows; i++) {
                 Object orderId = keyGenerateAlgorithm.generateKey();
-                result.add(new Object[]{orderId, generateInt(0, 1000), "'status'" + i, generateInt(-1000, 9999), generateInt(0, 100), generateInt(-64, 64), generateFloat(), generateDouble(),
-                        BigDecimal.valueOf(generateDouble()), false, generateString(6), "texts", "bytea".getBytes(), LocalDate.now(), LocalTime.now(), "2001-10-01",
+                result.add(new Object[]{orderId, generateInt(0, 1000), "'status'" + i, generateInt(-1000, 9999), generateInt(0, 100), generateFloat(), generateDouble(),
+                        BigDecimal.valueOf(generateDouble()), false, generateString(6), "texts", "bytea".getBytes(), new byte[]{0, 1, 2}, LocalDate.now(), LocalTime.now(), "2001-10-01",
                         Timestamp.valueOf(LocalDateTime.now()), OffsetDateTime.now(), "0 years 0 mons 1 days 2 hours 3 mins 4 secs", "{1, 2, 3}", generateJsonString(8, false),
                         generateJsonString(8, true), UUID.randomUUID().toString(), DigestUtils.md5Hex(orderId.toString()), "'rat' 'sat'", "tsquery", "0000", "[1,1000)",
                         "[2020-01-02,2021-01-01)", "[2020-01-01 00:00:00,2021-01-01 00:00:00)", "1 years 1 mons 10 days -06:00:00", "2000-01-02 00:00:00+00", "(1.0,1.0)", "[(0.0,0.0),(2.0,2.0)]",

--- a/test/e2e/operation/pipeline/src/test/java/org/apache/shardingsphere/test/e2e/data/pipeline/framework/helper/PipelineCaseHelper.java
+++ b/test/e2e/operation/pipeline/src/test/java/org/apache/shardingsphere/test/e2e/data/pipeline/framework/helper/PipelineCaseHelper.java
@@ -107,12 +107,13 @@ public final class PipelineCaseHelper {
         if (databaseType instanceof OpenGaussDatabaseType) {
             for (int i = 0; i < insertRows; i++) {
                 Object orderId = keyGenerateAlgorithm.generateKey();
+                byte[] bytesValue = {Byte.MIN_VALUE, -1, 0, 1, Byte.MAX_VALUE};
                 result.add(new Object[]{orderId, generateInt(0, 1000), "'status'" + i, generateInt(-1000, 9999), generateInt(0, 100), generateFloat(), generateDouble(),
-                        BigDecimal.valueOf(generateDouble()), false, generateString(6), "texts", "bytea".getBytes(), new byte[]{0, 1, 2}, LocalDate.now(), LocalTime.now(), "2001-10-01",
+                        BigDecimal.valueOf(generateDouble()), false, generateString(6), "texts", bytesValue, bytesValue, LocalDate.now(), LocalTime.now(), "2001-10-01",
                         Timestamp.valueOf(LocalDateTime.now()), OffsetDateTime.now(), "0 years 0 mons 1 days 2 hours 3 mins 4 secs", "{1, 2, 3}", generateJsonString(8, false),
-                        generateJsonString(8, true), UUID.randomUUID().toString(), DigestUtils.md5Hex(orderId.toString()), "'rat' 'sat'", "tsquery", "0000", "[1,1000)",
-                        "[2020-01-02,2021-01-01)", "[2020-01-01 00:00:00,2021-01-01 00:00:00)", "1 years 1 mons 10 days -06:00:00", "2000-01-02 00:00:00+00", "(1.0,1.0)", "[(0.0,0.0),(2.0,2.0)]",
-                        "(3.0,3.0),(1.0,1.0)", "<(5.0,5.0),5.0>", "1111", "192.168.0.0/16", "192.168.1.1", "08:00:2b:01:02:03", "\\x484c4c00000000002b05000000000000000000000000000000000000", 999});
+                        generateJsonString(8, true), UUID.randomUUID().toString(), DigestUtils.md5Hex(orderId.toString()), "'rat' 'sat'", "tsquery", "0000", "[1,1000)", "[2020-01-02,2021-01-01)",
+                        "[2020-01-01 00:00:00,2021-01-01 00:00:00)", "1 years 1 mons 10 days -06:00:00", "2000-01-02 00:00:00+00", "(1.0,1.0)", "[(0.0,0.0),(2.0,2.0)]", "(3.0,3.0),(1.0,1.0)",
+                        "<(5.0,5.0),5.0>", "1111", "192.168.0.0/16", "192.168.1.1", "08:00:2b:01:02:03", "\\x484c4c00000000002b05000000000000000000000000000000000000", 999});
             }
             return result;
         }

--- a/test/e2e/operation/pipeline/src/test/resources/env/scenario/general/opengauss.xml
+++ b/test/e2e/operation/pipeline/src/test/resources/env/scenario/general/opengauss.xml
@@ -29,6 +29,7 @@
         c_char character(32),
         c_text text,
         c_bytea bytea,
+        c_raw bytea,
         c_date date,
         c_time time without time zone,
         c_smalldatetime smalldatetime,
@@ -41,8 +42,11 @@
         c_uuid uuid,
         c_hash32 hash32,
         c_tsvector tsvector,
+        c_tsquery tsquery,
         c_bit bit(4),
         c_int4range int4range,
+        c_daterange daterange,
+        c_tsrange tsrange,
         c_reltime reltime,
         c_abstime abstime,
         c_point point,
@@ -54,16 +58,17 @@
         c_inet inet,
         c_macaddr macaddr,
         c_hll hll(14,10,12,0),
+        c_money money,
         PRIMARY KEY ( order_id )
         );
     </create-table-order>
 
     <full-insert-order>
         INSERT INTO test.t_order (
-        order_id, user_id, status, c_int, c_smallint, c_float, c_double, c_numeric, c_boolean, c_char, c_text, c_bytea, c_date, c_time,
-        c_smalldatetime, c_timestamp, c_timestamptz, c_interval, c_array, c_json, c_jsonb, c_uuid, c_hash32, c_tsvector, c_bit,
-        c_int4range, c_reltime, c_abstime, c_point, c_lseg, c_box, c_circle, c_bitvarying, c_cidr, c_inet, c_macaddr, c_hll
-        ) VALUES ( ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?);
+        order_id, user_id, status, c_int, c_smallint, c_float, c_double, c_numeric, c_boolean, c_char, c_text, c_bytea, c_raw, c_date, c_time,
+        c_smalldatetime, c_timestamp, c_timestamptz, c_interval, c_array, c_json, c_jsonb, c_uuid, c_hash32, c_tsvector, c_tsquery, c_bit,
+        c_int4range, c_daterange, c_tsrange, c_reltime, c_abstime, c_point, c_lseg, c_box, c_circle, c_bitvarying, c_cidr, c_inet, c_macaddr, c_hll, c_money
+        ) VALUES ( ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?);
     </full-insert-order>
 
     <create-table-order-item>

